src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -189,8 +192,10 @@ private void validateName(final String name, final EventTypeBase eventType) thro
 
     private void validateSchema(final EventTypeBase eventType) throws InvalidEventTypeException {
         try {
-            final JSONObject schemaAsJson = new JSONObject(eventType.getSchema().getSchema());
+            final String eventTypeSchema = eventType.getSchema().getSchema();
+            checkJsonIsValid(eventTypeSchema);
 
+            final JSONObject schemaAsJson = new JSONObject(eventTypeSchema);
```

Comment:
Looking at https://github.com/everit-org/json-schema#quickstart you can create a `JSONObject` from a `JSONTokener`, you could then call `next` or `more` on the tokener to determine if it reached the end of the input.